index.js

```diff
@@ -64,25 +55,47 @@ function style (raw) {
   var len = fields.length;
   var i = -1;
   var s;
+  var field;
 
   while (++i < len) {
-    s = fields[i].indexOf(':');
-    result[fields[i].slice(0, s)] = fields[i].slice(s + 1).trim();
+    field = fields[i].trim();
+    if (!field) continue;
+    s = field.indexOf(':');
+    result[field.slice(0, s)] = field.slice(s + 1).trim();
   }
 
   return result;
 }
 
-function createDataSet (props) {
-  var dataset;
-  var key;
-
-  for (key in props) {
-    if (key.slice(0, 5) == 'data-') {
-      dataset || (dataset = {});
-      dataset[camel(key.slice(5))] = props[key];
+function createProperties (attributes) {
+  var properties = {};
+
+  for (var key in attributes) {
+    if (!attributes.hasOwnProperty(key)) continue;
+
+    switch (key) {
+    case 'style':
+      properties.style = style(attributes.style);
+      break;
+    case 'class':
+      properties.className = attributes['class'];
+      break;
+    case 'for':
+      properties.htmlFor = attributes['for'];
+      break;
+    default:
+      if (key.indexOf('data-') === 0) {
+        addDataSet(properties, key, attributes[key]);
+      } else {
+        properties[key] = attributes[key];
```

Comment:
In order to support custom attributes, replace this line with:

`attrs[key]` = attributes[key];